contrib/IECoreUSD/src/IECoreUSD/MeshAlgo.cpp

```diff
@@ -136,4 +136,60 @@ bool meshMightBeTimeVarying( pxr::UsdGeomMesh &mesh )
 
 ObjectAlgo::ReaderDescription<pxr::UsdGeomMesh> g_curvesReaderDescription( pxr::TfToken( "Mesh" ), readMesh, meshMightBeTimeVarying );
 
+} // namespace
+
+//////////////////////////////////////////////////////////////////////////
+// Writing
+//////////////////////////////////////////////////////////////////////////
+
+namespace
+{
+
+void writeMesh( const IECoreScene::MeshPrimitive *mesh, const pxr::UsdStagePtr &stage, const pxr::SdfPath &path, pxr::UsdTimeCode time )
+{
+	auto usdMesh = pxr::UsdGeomMesh::Define( stage, path );
+
+	// Topology
+
+	usdMesh.CreateFaceVertexCountsAttr().Set( DataAlgo::toUSD( mesh->verticesPerFace() ), time );
+	usdMesh.CreateFaceVertexIndicesAttr().Set( DataAlgo::toUSD( mesh->vertexIds() ), time );
+
+	// Interpolation
+
+	if( mesh->interpolation() == std::string( "catmullClark" ) )
+	{
+		usdMesh.CreateSubdivisionSchemeAttr().Set( pxr::UsdGeomTokens->catmullClark );
+	}
+	else
+	{
+		usdMesh.CreateSubdivisionSchemeAttr().Set( pxr::UsdGeomTokens->none );
+	}
+
+	// Corners
+
+	if( mesh->cornerIds()->readable().size() )
+	{
+		usdMesh.CreateCornerIndicesAttr().Set( DataAlgo::toUSD( mesh->cornerIds() ) );
+		usdMesh.CreateCornerSharpnessesAttr().Set( DataAlgo::toUSD( mesh->cornerSharpnesses() ) );
+	}
+
+	// Creases
+
+	if( mesh->creaseLengths()->readable().size() )
+	{
+		usdMesh.CreateCreaseLengthsAttr().Set( DataAlgo::toUSD( mesh->creaseLengths() ) );
+		usdMesh.CreateCreaseIndicesAttr().Set( DataAlgo::toUSD( mesh->creaseIds() ) );
+		usdMesh.CreateCreaseSharpnessesAttr().Set( DataAlgo::toUSD(  mesh->creaseSharpnesses() ) );
```

Comment:
We read these time varying, and as we support writing varying topology, do we need `time` in these `Set` calls too?

Reply:
Fixed in 3b973a5. Note that I didn't update the writing for the subdivision scheme, as that is documented as not being animatable (that is what `SdfVariabilityUniform` means, right?).

Reply:
Perfect, that's my understanding of it.